homeassistant/config_entries.py

```diff
@@ -828,6 +843,53 @@ async def _async_init_reauth(
             translation_key="config_entry_reauth",
         )
 
+    @callback
+    def async_start_reconfigure(
+        self,
+        hass: HomeAssistant,
+        context: dict[str, Any] | None = None,
+        data: dict[str, Any] | None = None,
+    ) -> None:
+        """Start a reconfigure flow."""
+        # We will check this again in the task when we hold the lock,
+        # but we also check it now to try to avoid creating the task.
+        if any(self.async_get_active_flows(hass, {SOURCE_RECONFIGURE})):
+            # Reconfigure flow already in progress for this entry
+            return
+        if any(self.async_get_active_flows(hass, {SOURCE_REAUTH})):
+            # Reauth flow already in progress for this entry
+            return
```

Comment:
Instead of just return maybe we should kick in the reauth flow?

Reply:
That seems like a job for the frontend?

Reply:
Makes sense. Probably we should not show the option to reconfigure if an active "reauth" exists